src/AutoRest.CSharp/Common/Output/Models/Types/ExpressionTypeProvider.cs

```diff
@@ -12,29 +13,52 @@ namespace AutoRest.CSharp.Output.Models.Types
     // TODO -- eventually we should combine everything in this class into TypeProvider
     internal abstract class ExpressionTypeProvider : TypeProvider
     {
-        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel) : base(defaultNamespace, sourceInputModel)
+        protected ExpressionTypeProvider(string defaultNamespace, SourceInputModel? sourceInputModel)
+            : base(defaultNamespace, sourceInputModel)
         {
+            DeclarationModifiers = ClassSignatureModifiers.Partial | ClassSignatureModifiers.Public;
         }
 
         private IReadOnlyList<string>? _usings;
         public IReadOnlyList<string> Usings => _usings ??= BuildUsings().ToArray();
 
+        public ClassSignatureModifiers DeclarationModifiers { get; init; }
```

Comment:
I think we should make it `protected init;` because we never want to accidentally assign the value of this when new an derived type of this

Reply:
I think we forgot to change this?